solidity/test/VendingMachineTest.js

```diff
@@ -768,8 +768,92 @@ describe("VendingMachine", async function() {
+
+      await increaseTime(7 * 24 * 60 * 60 - 10 * 60) // 7 days minus 10 minutes
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(1000))
+
+      await increaseTime(60 * 60) // one hour
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.not.eq.BN(btcToTbtc(1000))
+    })
+
+    it("has a max supply of 1500 BTC between the 35th day and the 42nd day", async () => {
+      await increaseTime(35 * 24 * 60 * 60 + 1) // 35 days and 1 second
+      let maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(1500))
+
+      await increaseTime(7 * 24 * 60 * 60 - 10 * 60) // 7 days minus 10 minutes
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(1500))
+
+      await increaseTime(60 * 60) // one hour
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.not.eq.BN(btcToTbtc(1500))
+    })
+
+    it("has a max supply of 2000 BTC between the 42nd day and the 49th day", async () => {
+      await increaseTime(42 * 24 * 60 * 60 + 1) // 42 days and 1 second
+      let maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(2000))
+
+      await increaseTime(7 * 24 * 60 * 60 - 10 * 60) // 7 days minus 10 minutes
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(2000))
+
+      await increaseTime(60 * 60) // one hour
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.not.eq.BN(btcToTbtc(2000))
+    })
+
+    it("has a max supply of 2500 BTC between the 49th day and the 56th day", async () => {
+      await increaseTime(49 * 24 * 60 * 60 + 1) // 42 days and 1 second
+      let maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(2500))
+
+      await increaseTime(7 * 24 * 60 * 60 - 10 * 60) // 7 days minus 10 minutes
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(2500))
+
+      await increaseTime(60 * 60) // one hour
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.not.eq.BN(btcToTbtc(2500))
+    })
+
+    it("has a max supply of 3000 BTC between the 56th day and the 63rd day", async () => {
+      await increaseTime(56 * 24 * 60 * 60 + 1) // 42 days and 1 second
+      let maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(3000))
+
+      await increaseTime(7 * 24 * 60 * 60 - 10 * 60) // 7 days minus 10 minutes
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.eq.BN(btcToTbtc(3000))
+
+      await increaseTime(60 * 60) // one hour
+      maxSupply = await vendingMachine.getMaxSupply()
+      expect(maxSupply).to.not.eq.BN(btcToTbtc(3000))
+    })
+
+    it("has a max supply of 21M BTC after the 63rd day", async () => {
```

Comment:
It really feels like it's time to make this a table test, tbh.

Reply:
Yeah. I've been sticking with "KISS" at the expense of DRY here in a misguided effort to stave off mistakes... but this has gotten ridiculous

Reply:
Yeah, was pretty much on the same page in the first iteration, this round made me reconsider :grin:

Reply:
I would embrace the Go approach here tbh, something like:

```
const cases = [
  { start: 0, end: 2, maxSupply: 2 },
  { start: 2, end: 7, maxSupply: 100 },
  ...
]

for (const { start, end, maxSupply } in cases) {
  it(`has a max supply of ${maxSupply} between day ${start} and day ${end}`, async () => {
    await increaseTime(start * DAYS)
    expect(await vendingMachine.getMaxSupply()).to.eq.BN(btcToTbtc(maxSupply))

    await increaseTime((end - start) * DAYS - 10 * MINUTES)
    expect(await vendingMachine.getMaxSupply()).to.eq.BN(btcToTbtc(maxSupply))

    await increaseTime(1 * 60 * MINUTES)
    expect((await vendingMachine.getMaxSupply()).gtn(btcToTbtc(maxSupply))).to.be.true
  })
}
```

Ish.